common/apps/apps.go

```diff
@@ -0,0 +1,140 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+package apps
+
+import (
+	"github.com/coreos/rocket/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+)
+
+type App struct {
+	image string   // the image as-supplied by the user on the cli
+	args  []string // any arguments the user supplied for this app
+	asc   string   // signature file override for image verification (if fetching occurs)
+
+	imageID types.Hash // resolved image identifier
+}
+
+type Apps struct {
+	apps []App
+}
+
+// Reset creates a new slice for al.apps, needed by tests
+func (al *Apps) Reset() {
+	al.apps = make([]App, 0)
+}
+
+// Count returns the number of apps in al
+func (al *Apps) Count() int {
+	return len(al.apps)
+}
+
+// Create creates a new app in al and returns a pointer to it
+func (al *Apps) Create(img string) {
+	al.apps = append(al.apps, App{image: img})
+}
+
+// Last returns a pointer to the top app in al
+func (al *Apps) Last() *App {
+	if len(al.apps) == 0 {
+		return nil
+	} else {
+		return &al.apps[len(al.apps)-1]
+	}
+}
+
+// Walk iterates on al.apps calling f for each app
+// walking stops if f returns an error, the error is simply returned
+func (al *Apps) Walk(f func(*App) error) error {
+	for i, _ := range al.apps {
+		// FIXME(vc): is there a way to make:
+		//  for _, a := range al.apps
+		// produce a pointer to the app in 'a'?
+		a := &al.apps[i]
+		if err := f(a); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// AppendArg appends another argument onto the app
+func (a *App) AppendArg(arg string) {
+	a.args = append(a.args, arg)
+}
+
+// GetArgs gets the args for a given app
+func (a *App) GetArgs() []string {
+	return a.args
+}
+
+// GetImage gets the image for a given app
+func (a *App) GetImage() string {
+	return a.image
+}
+
+// SetImageID sets the image ID for a given app
+func (a *App) SetImageID(imageID types.Hash) {
+	a.imageID = imageID
+}
+
+// GetImageID gets the image ID for a given app
+func (a *App) GetImageID() types.Hash {
+	return a.imageID
+}
+
+// SetAsc sets the ascii-armored signature file path for the app
+func (a *App) SetAsc(asc string) {
+	a.asc = asc
+}
+
+// GetAsc gets the ascii-armored signature file path for the app
+func (a *App) GetAsc() string {
+	return a.asc
+}
+
+// these convenience functions just return typed lists containing just the named member
+// TODO(vc): these probably go away when we just pass Apps to stage0
+
+// getImages returns a list of the images in al, one per app.
```

Comment:
nit, capitalisation of docstrings here and below is inconsistent with actual function names